zebra-chain/src/types.rs

```diff
@@ -30,6 +35,47 @@ impl fmt::Debug for Sha256dChecksum {
 #[derive(Copy, Clone, Debug, Eq, PartialEq)]
 pub struct BlockHeight(pub u32);
 
+/// A Bitcoin-style `locktime`, representing either a block height or an epoch
+/// time.
+///
+/// # Invariants
+///
+/// Users should not construct a `LockTime` with `BlockHeight` greater than or
+/// equal to `500_000_000` or a timestamp before 4 November 1985 (Unix timestamp
+/// less than `500_000_000`).
```

Comment:
This seems like "good enough" data modeling rather than trying to ensure that the invariant is upheld by making an opaque struct with constructors, getters, etc; this way, users can match on a `LockTime`.